collator/src/collator/do_collate/finalize.rs

```diff
@@ -97,6 +108,8 @@ impl Phase<FinalizeState> {
         )
         .serialize();
 
+        let processed_upto = diff_with_messages.processed_upto.clone();
```

Comment:
Let's rename this into `diff_processed_to` across collation methods to separate from real `processed_upto { processed_to, read_to }`.